onnxruntime/core/providers/xnnpack/nn/qlinear_conv.cc

```diff
@@ -0,0 +1,289 @@
+      static_cast<uint32_t>(conv_attrs.strides[0]),
+      static_cast<uint32_t>(conv_attrs.strides[1]),
+      static_cast<uint32_t>(conv_attrs.dilations[0]),
+      static_cast<uint32_t>(conv_attrs.dilations[1]),
+      static_cast<uint32_t>(group_count),
+      static_cast<size_t>(group_input_channels),
+      static_cast<size_t>(group_output_channels),
+      static_cast<size_t>(C),
+      static_cast<size_t>(M),
+      quant_param.X_zero_point_value, quant_param.X_scale_value,
+      quant_param.W_zero_point_value, quant_param.W_scale_value,
+      W.Data<uint8_t>(), B_data,
+      quant_param.Y_zero_point_value, quant_param.Y_scale_value,
+      output_min, output_max,
+      0,  // flags
+#ifdef XNN_CACHE_ENABLE
+      caches_t,
+#endif
+      &p);
+  if (status != xnn_status_success) {
+    return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL,
+                           "Failed to create xnnpack kernel. xnn_create_convolution2d_nhwc_qu8 returned ", status);
+  }
+  return Status::OK();
+}
+}  // namespace
+
+class QLinearConv : public OpKernel {
+ public:
+  explicit QLinearConv(const OpKernelInfo& info) : OpKernel(info),
+                                                   conv_attrs_(info) {
+
+    const auto& node{Node()};
+
+    const auto& input_defs = node.InputDefs();
+    const NodeArg& X = *input_defs[0];
+    C_ = X.Shape()->dim(3).dim_value();  // input is NHWC. op support checker made sure C dim was known
+
+    // as the weight input is a constant initializer we can calculate all the sizes here instead of in Compute
+    const Tensor* W = nullptr;
+    ORT_ENFORCE(info.TryGetConstantInput(InputTensors::IN_W, &W),
+                "Weight input was not constant initializer. XNNPACK EP should not have asked for the node. Node name:",
+                node.Name());
+    //quant param, which used in create xnnpack_conv_kernel
+    const Tensor* X_zero_point = nullptr;
+    const Tensor* W_zero_point = nullptr;
+    const Tensor* Y_zero_point = nullptr;
+    ORT_ENFORCE(info.TryGetConstantInput(InputTensors::IN_X_ZERO_POINT,&X_zero_point),
+        "X_zero_point input was not constant initializer. XNNPACK EP should not have asked for the node. Node name:",
+                node.Name());
+    ORT_ENFORCE(info.TryGetConstantInput(InputTensors::IN_W_ZERO_POINT, &W_zero_point),
+                "W_zero_point input was not constant initializer. XNNPACK EP should not have asked for the node. Node name:",
+                node.Name());
+    ORT_ENFORCE(info.TryGetConstantInput(InputTensors::IN_Y_ZERO_POINT, &Y_zero_point),
+                "Y_zero_point input was not constant initializer. XNNPACK EP should not have asked for the node. Node name:",
+                node.Name());
+
+    quant_param_.X_zero_point_value = *(X_zero_point->template Data<uint8_t>());
+    quant_param_.W_zero_point_value = *(W_zero_point->template Data<uint8_t>());
+    quant_param_.Y_zero_point_value = *(Y_zero_point->template Data<uint8_t>());
```

Comment:
I will remove all qlinear_xx.cc in xnnpack ep, because, the type is just a condition for different create/setup functions, so we can reuse a lot general logic an save binary size.